poly-commitment/src/commitment.rs

```diff
@@ -900,18 +897,67 @@ mod tests {
                 let mut e = vec![Fp::zero(); n];
                 e[i] = Fp::one();
                 let p = Evaluations::<Fp, D<Fp>>::from_vec_and_domain(e, domain).interpolate();
-                let c = srs.commit_non_hiding(&p, None);
-                assert!(c.shifted.is_none());
-                assert_eq!(c.unshifted.len(), 1);
-                c.unshifted[0]
+                srs.commit_non_hiding(&p, None)
+            })
+            .collect();
+
+        let computed_lagrange_commitments = srs.lagrange_bases.get(&domain.size()).unwrap();
+        for i in 0..n {
+            assert_eq!(
+                computed_lagrange_commitments[i],
+                expected_lagrange_commitments[i],
+            );
+        }
+    }
+
+    #[test]
+    fn test_chunked_lagrange_commitments() {
+        let n = 64;
+        let domain = D::<Fp>::new(n).unwrap();
+
+        let mut srs = SRS::<VestaG>::create(n / 2);
+        srs.add_lagrange_basis(domain);
+
+        let expected_lagrange_commitments: Vec<_> = (0..n)
+            .map(|i| {
+                let mut e = vec![Fp::zero(); n];
+                e[i] = Fp::one();
+                let p = Evaluations::<Fp, D<Fp>>::from_vec_and_domain(e, domain).interpolate();
+                srs.commit_non_hiding(&p, None)
+            })
+            .collect();
+
+        let computed_lagrange_commitments = srs.lagrange_bases.get(&domain.size()).unwrap();
+        for i in 0..n {
+            assert_eq!(
+                computed_lagrange_commitments[i],
+                expected_lagrange_commitments[i],
```

Comment:
BTW I'm still trying to understand why the redundancy in the lagrange basis. Here your lagrange base will have the same basis written twice (each half being the same):

```
computed_lagrange_commitments[n / 2] == computed_lagrange_commitments[0]
```

why do this instead of modifying the algorithm? As this is a waste of space no?

Reply:
This assertion is untrue. You can confirm yourself by injecting it into the test
```rust
assert_eq!(computed_lagrange_commitments[n / 2], computed_lagrange_commitments[0]);
```

Reply:
oh yeah I think I meant

```
computed_lagrange_commitments[0].unshifted[0] == computed_lagrange_commitments[0].unshifted[1]
```

Reply:
mm, for some reason it's not always true for all index